pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -590,6 +614,16 @@ func insertOrUpdateCondition(d data.Object, desiredCondition summary.Condition)
 	return true, nil
 }
 
+func getCondition(d data.Object, conditionType string) *summary.Condition {
```

Comment:
Could use this func in `shouldForceRemoval` implementation.

Reply:
Yes, good point. I will update the implementation.